larpandoracontent/LArHelpers/LArFormattingHelper.cc

```diff
@@ -0,0 +1,184 @@
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    LArFormattingHelper::Table::Table(const std::vector<std::string> &columnTitles, const unsigned int &precision) :
+        m_columnTitles(columnTitles),
+        m_precision(precision)
+    {
+        m_stringstream.precision(m_precision);
+
+        for (const std::string &title : m_columnTitles)
+            m_widths.push_back(title.length());
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::CheckForSeparatorColumn()
+    {
+        if (m_columnTitles[m_elements.size() % m_columnTitles.size()] == "")
+        {
+            m_format.push_back("");
+            m_elements.push_back("");
+        }
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    bool LArFormattingHelper::Table::IsSeparatorColumn(const unsigned int &column)
+    {
+        if (column>=m_columnTitles.size())
+            throw StatusCodeException(STATUS_CODE_OUT_OF_RANGE);
+
+        return (m_columnTitles[column] == "");
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::UpdateColumnWidth()
+    {
+        const unsigned int currentElementIndex(m_elements.size()-1);
+        const unsigned int column(currentElementIndex % m_widths.size());
+        m_widths[column] = std::max(static_cast<unsigned int>(m_elements[currentElementIndex].length()), m_widths[column]);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+   
+    void LArFormattingHelper::Table::Print()
+    {
+        if (m_columnTitles.size() == 0)
+            return;
+
+        this->PrintColumnTitles();
+        this->PrintHorizontalLine();
+        this->PrintTableElements();
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::PrintColumnTitles()
+    {
+        for (unsigned int i=0; i<m_columnTitles.size(); i++)
+            this->PrintTableCell(m_columnTitles[i], LArFormattingHelper::GetFormatCharacter(LArFormattingHelper::BOLD), i);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::PrintHorizontalLine()
+    {
+        for (unsigned int i=0; i<m_columnTitles.size(); i++) 
```

Comment:
See comments above. Note that checking size of m_columnTitles repeatedly here, but this might be safest. This isn't likely to be performance-critical code.

Reply:
m_columnTitles is const, so replaced with

     const unsigned int nColumns(m_columnTitles.size());                                                                                      
     for (unsigned int i = 0; i < nColumns; ++i)

Reply:
`m_columnTitles` is `const`, so replaced with
```c++
const unsigned int nColumns(m_columnTitles.size());                                                                                      
for (unsigned int i = 0; i < nColumns; ++i)  
```